From 183d5a36e3591a42400168387235c5fac1794201 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:27:38 -0400
Subject: [alsa] hda: Use analog beep for Thinkpads with AD1984 codecs

Message-id: <140972823573-228-patchset-jkysela@redhat.com>
Patchwork-id: 90928
O-Subject: [RHEL 7.1 PATCH 227/442] ALSA: hda - Use analog beep for Thinkpads with AD1984 codecs
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

For making the driver behavior compatible with the earlier kernels,
use the analog beep in the loopback path instead of the digital beep.

Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: f3e9b59cb948e2328bc06635ad39572d5b7b4791)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_analog.c b/sound/pci/hda/patch_analog.c
index fff09be..40ba06e 100644
--- a/sound/pci/hda/patch_analog.c
+++ b/sound/pci/hda/patch_analog.c
@@ -1025,6 +1025,9 @@ static void ad1884_fixup_thinkpad(struct hda_codec *codec,
   spec->gen.keep_eapd_on = 1;
   spec->gen.vmaster_mute.hook = ad_vmaster_eapd_hook;
   spec->eapd_nid = 0x12;
+  /* Analog PC Beeper - allow firmware/ACPI beeps */
+  spec->beep_amp = HDA_COMPOSE_AMP_VAL(0x20, 3, 3, HDA_INPUT);
+  spec->gen.beep_nid = 0; /* no digital beep */
  }
 }
 
-- 
1.7.1